app/views/feeds/_form_apple_config.html.erb

```diff
@@ -6,38 +6,66 @@
     <div class="card-body" data-controller="apple-key">
       <div class="mb-4">
         <p><%= t(".description") %></p>
-        <a href="https://help.prx.org/hc/en-us/articles/26602803552155-Setting-up-Apple-Podcast-Subscriptions-with-Apple-Delegated-Delivery"><%= t(".guide_link") %></a>
+        <%= t(".guide_link").html_safe %>
       </div>
       <%= form.fields_for :apple_config do |config_form| %>
         <%= config_form.fields_for :key do |key_fields| %>
           <% unless config_form.object.persisted? %>
-            <%= key_fields.file_field :key_file, class: "mb-4 form-control", accept: ".p8, .pem", data: {action: "apple-key#convertFileToKey"} %>
+            <%= key_fields.file_field :key_file, class: "mb-4 form-control", accept: ".p8, .pem", data: {action: "apple-key#convertFileToKey"}, required: true %>
 
             <%= key_fields.hidden_field :key_pem_b64, data: {apple_key_target: "pem"} %>
+          <% else %>
+            <div class="mb-2"><%= t(".key_uploaded") %></div>
+            <div class="mb-4 form-floating">
+              <%= key_fields.text_field :provider_id, disabled: true, data: {apple_key_target: "provider"} %>
+              <%= key_fields.label :provider_id, "Provider ID" %>
+            </div>
+            <div class="form-floating">
+              <%= key_fields.text_field :key_id, disabled: true, data: {apple_key_target: "key"} %>
+              <%= key_fields.label :key_id, "Apple Key" %>
+            </div>
           <% end %>
-          <div class="mb-4 form-floating">
-            <%= key_fields.text_field :provider_id, disabled: true, data: {apple_key_target: "provider"} %>
-            <%= key_fields.label :provider_id, "Provider ID" %>
+          <%= key_fields.hidden_field :provider_id, data: {apple_key_target: "provider"} %>
+          <%= key_fields.hidden_field :key_id, data: {apple_key_target: "key"} %>
+        <% end %>
+        <% if config_form.object.persisted? %>
```

Comment:
within this form, also hides all the other fields until a valid key is uploaded and saved